api/mapping/services.py

```diff
@@ -174,26 +179,97 @@ def process_scan_report(scan_report_id):
+                definition_fixed=False,
+            )
 
-def import_data_dictionary(filepath):
-    filepath = filepath
 
-    with open(filepath, 'rt') as f:
-        reader = csv.reader(f)
-        next(reader)  # Skip header row
+def build_usagi_index():
+    # Use 'build' to create index for the first time.
+    # Index stored in data/usagi/mainIndex
+    p = subprocess.Popen(
+        "java -jar Usagi.jar build usagi_build_index.properties",
+        cwd="/data/usagi",
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    )
 
-        # Assumes that the input columns are in the same order as the Twins data dictionary
-        for row in reader:
+    # Print console output to screen so you can see it's working (or not!)
+    stdout = []
+    while True:
+        line = p.stdout.readline()
+        if not isinstance(line, (str)):
+            line = line.decode("utf-8")
+        stdout.append(line)
+        print(line)
+        if line == "" and p.poll() != None:
+            break
+
+
+def run_usagi(scan_report_id):
+
+    print("RUNNING USAGI....")
+
+    # Grab data from DataDictionary
+    # This filters out all entries in the dictionary
+    # which are marked PatientID/DateEvent/Ignore
+    dict = (
+        DataDictionary.objects.filter(
+            source_value__scan_report_field__scan_report_table__scan_report__id=scan_report_id
+        )
+        .filter(source_value__scan_report_field__is_patient_id=False)
+        .filter(source_value__scan_report_field__is_date_event=False)
+        .filter(source_value__scan_report_field__is_ignore=False)
+        .exclude(source_value__value='List truncated...')
+    )
 
-            if row[0][0] == '':
-                continue
+    dict_df = pd.DataFrame.from_dict(
+        dict.values(
+            "source_value__scan_report_field__scan_report_table__scan_report__data_partner__name",
+            "source_value__scan_report_field__scan_report_table__scan_report__dataset",
+            "source_value__scan_report_field__scan_report_table__name",
+            "source_value__scan_report_field__name",
+            "source_value__value",
+            "source_value__frequency",
+            "dictionary_field_description",
+            "dictionary_value_description",
+        )
+    )
 
-            print(row)
-            
-            DataDictionary.objects.create(
-                table=row[0],
-                field=row[1],
-                field_description=row[2],
-                value_code=row[3],
-                value_description=row[4]
-            )
\ No newline at end of file
+    # Name columns
+    dict_df.columns = [
+        "DataPartner",
+        "DataSet",
+        "Table",
+        "Field",
+        "Value",
+        "Frequency",
+        "FieldDesc",
+        "ValueDescription",
+    ]
+
+    df = dict_df
+
+    # df = df.head(10)
+    df.to_csv("/data/usagi/input/usagi_input_data.csv", index=False)
```

Comment:
What's the likelihood of two instances of Usagi being run at the same time by different users?
I.e. would it be safer to have the report name/id in the temp save file so there can't be overwriting issues?

Suggestion:
```
df.to_csv(f"/data/usagi/input/usagi_input_data_{scan_report_id}.csv", index=False)
```

Reply:
Reasonably likely I would say!

Not sure if I mentioned earlier in the Dev team meeting - this PR really is pretty ropey but I was keen to get the code into master so I can start dishing out smaller tasks to the team. If I wait until all the functionality is done, the merge would be a mightmare, I think.

Reply:
I should add, the functionality in this PR can't actually be accessed/run by the app users (unless they know specific URLs to call the methods e.g. http://localhost:8080/datadictionary/merge/?search=1 and http://localhost:8080/testusagi/1/)

Reply:
cool well i'll let you add is as follow up task then